libs/product/src/selectors/composite-product/composite-product.selectors.ts

```diff
@@ -79,94 +78,97 @@ const createCompositeProductSelectors = (): DaffCompositeProductMemoizedSelector
 	);
 
 	/**
-	 * Selector for whether the composite product could have a price range.
+	 * Selector for whether the composite product could have a price range in any configuration,
+	 * including the optional items and regardless of the current item option selection.
+	 * This could be useful for a quick preview of the product.
 	 */
-	const selectCompositeProductPossiblyHasPriceRange = createSelector(
+	const selectCompositeProductHasOptionalItemPriceRange = createSelector(
 		selectProductEntities,
-		(products, props) => selectCompositeProductMinPossiblePrice.projector(products, { id: props.id }) 
-			!== selectCompositeProductMaxPossiblePrice.projector(products, { id: props.id })
+		(products, props) => selectCompositeProductMinOptionalItemPrice.projector(products, { id: props.id }) 
+			!== selectCompositeProductMaxOptionalItemPrice.projector(products, { id: props.id })
 	)
 
 	/**
-	 * Selector for the minimum price for required items of a composite product, depending on the current selection of item options.
-	 * This would be used for a composite product that has required items without default selections.
+	 * Selector for the minimum discounted price for a composite product, including optional items and regardless of the current selection of item options.
+	 * This could be useful for a quick preview of the product.
 	 */
-	const selectCompositeProductMinPrice = createSelector(
+	//todo use optional chaining where possible.
+	const selectCompositeProductMinOptionalItemDiscountedPrice = createSelector(
 		selectProductEntities,
-		selectCompositeProductAppliedOptionsEntitiesState,
-		(products, appliedOptionsEntities, props) => {
+		(products, props) => {
 			const product = selectProduct.projector(products, { id: props.id });
 			if(product.type !== DaffProductTypeEnum.Composite) {
 				return undefined;
 			}
-			const appliedOptions = selectCompositeProductAppliedOptions.projector(products, appliedOptionsEntities, { id: props.id });
 			return (<DaffCompositeProduct>product).items.reduce((acc, item) => daffAdd(
 				acc, 
-				appliedOptions[item.id] ? daffMultiply(appliedOptions[item.id].price, appliedOptions[item.id].quantity) : getMinimumCompositeItemPrice(item)
-			), product.price);
+				Math.min(...item.options.map(option => daffSubtract(option.price, option.discount ? option.discount.amount : 0)))
```

Comment:
I made it a function. It isn't that straightforward to extract it into a selector, because I would need to make another selector that would select particular items off of state so I could just project the item here into the first selector. Like this:
```
const selectCompositeProductItemDiscountedPrices = createSelector(
   selectCompositeProductItem,     <----------------------------- this needs to be defined too.
   (item, props) => selectorStuff
)
```
So I don't think it's worth it